src/superface-test.utils.ts

```diff
@@ -0,0 +1,243 @@
+      superJsonNormalized.profiles[profileId].providers[providerId];
+
+    if (!('file' in targetedProfileProvider)) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+/**
+ * Returns profile id if entered profile is either instance of Profile or string
+ */
+export function getProfileId(profile: Profile | string): string {
+  if (profile instanceof Profile) {
+    return profile.configuration.id;
+  }
+
+  return profile;
+}
+
+/**
+ * Returns provider id if entered provider is either instance of Provider or string
+ */
+export function getProviderName(provider: Provider | string): string {
+  if (provider instanceof Provider) {
+    return provider.configuration.name;
+  }
+
+  return provider;
+}
+
+/**
+ * Returns usecase name if entered usecase is either instance of UseCase or string
+ */
+export function getUseCaseName(useCase: UseCase | string): string {
+  if (useCase instanceof UseCase) {
+    return useCase.name;
+  }
+
+  return useCase;
+}
+
+/**
+ * Returns SuperJson based on path detected with its abstract method.
+ */
+export async function getSuperJson(): Promise<SuperJson> {
+  const superPath = await SuperJson.detectSuperJson(process.cwd(), 3);
+
+  if (superPath === undefined) {
+    throw new SuperJsonNotFoundError();
+  }
+
+  return (await SuperJson.load(joinPath(superPath, 'super.json'))).unwrap();
+}
+
+const HIDDEN_CREDENTIALS_PLACEHOLDER =
+  'credentials-removed-to-keep-them-secure';
+const AUTH_HEADER_NAME = 'Authorization';
+
+export function assertsRecordingsAreNotStrings(
+  recordings: string[] | RecordingDefinition[]
+): asserts recordings is RecordingDefinition[] {
+  for (const recording of recordings) {
+    if (typeof recording === 'string') {
+      throw new UnexpectedError('recording is a string, not object');
+    }
+  }
+}
+
+export async function removeSensitiveInformation(
+  sfConfig: CompleteSuperfaceTestConfig,
+  recordings: RecordingDefinition[]
+): Promise<void> {
+  const {
+    configuration: { security, baseUrl },
+  } = await sfConfig.client.cacheBoundProfileProvider(
+    sfConfig.profile.configuration,
+    sfConfig.provider.configuration
+  );
+
+  for (const recording of recordings) {
+    for (const scheme of security) {
+      removeCredentialsBasedOnScheme(recording, scheme, baseUrl);
+    }
+  }
+}
+
+export function removeCredentialsBasedOnScheme(
+  recording: RecordingDefinition,
+  scheme: SecurityScheme,
+  baseUrl: string
+): void {
+  if (isApiKeySecurityScheme(scheme)) {
+    if (scheme.in === ApiKeyPlacement.HEADER) {
+      if (recording.reqheaders?.[scheme.name] !== undefined) {
+        recording.reqheaders[scheme.name] = HIDDEN_CREDENTIALS_PLACEHOLDER;
+      }
+    } else if (scheme.in === ApiKeyPlacement.BODY) {
+      if (
+        typeof recording.body === 'object' &&
+        !(recording.body instanceof RegExp) &&
+        !(recording.body instanceof Buffer) &&
+        !Array.isArray(recording.body) &&
+        !('includes' in recording.body) &&
```

Comment:
Cursed. Can you please add a comment here so that other poor souls know the pain of detecting an array which is not an array but is an array?